### PR TITLE
Cut v0.51.0

### PR DESCRIPTION
**Pull request by simonpasquier**
## Description

Prepare the v0.51.0 release
- [X] Bump Prometheus and Alertmanager versions
- [X] Bump Go dependencies
- [X] Update changelog
- [x] #4272 merged
- [x] #4275 merged
- [x] #4276 merged

## Type of change

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [X] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

```release-note

```

### CODE REVIEW

**Comment by simonpasquier** [comment]
@prometheus-operator/prometheus-operator-reviewers for review

**Comment by simonpasquier** [comment]
@PhilipGough, @fpetkovski  and @slashpai: it would be nice if you can review it too :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,18 @@
+## 0.51.0 / 2021-09-22
+
+* [FEATURE] Add `metricRelabelings` field to the Probe CRD for configuring the metric relabel configs. #4226
```
Comment (slashpai) [inline CHANGELOG.md:3]: Should this be like `Added` instead of `Add` ?

**Comment by slashpai** [comment]
Otherwise lgtm :)

File: CHANGELOG.md (hunk 1 above)
Comment (simonpasquier) [reply CHANGELOG.md]: yes I agree!

**Review by slashpai — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,18 @@
+## 0.51.0 / 2021-09-24
+
+* [FEATURE] Added `metricRelabelings` field to the Probe CRD for configuring the metric relabel configs. #4226
+* [FEATURE] Added `volumeMounts` field to the Prometheus CRD for configuring the volume mounts of the thanos-sidecar container. #4238
+* [FEATURE] Added  `enforcedBodySizeLimit` field to the Prometheus CRD. #4275
```
Comment (PhilipGough) [inline CHANGELOG.md:5]: nit: extra whitespace

**Review by PhilipGough — APPROVED** [review]
lgtm on green

File: CHANGELOG.md (hunk 2 above)
Comment (simonpasquier) [reply CHANGELOG.md]: cool, thanks!

**Pull request merged**